Moving boxes example thorn

lucass-carneiro opened the pull request:
This PR provides a new thorn called `MovingBoxesToy` that sets up two refinement boxes and makes them move around in a circle. It is intended as a guide for people wanting to implement more complex moving boxes.

In order to make this thorn possible, I had to change the parameters in `BoxInBox` to `RESTRICTED`

eschnett commented:
Thanks! This thorn is a good idea.

I don't think that `BoxInBox`'s parameters should be "restricted", i.e. should be visible outside the thorn. If `MovingBoxToy` wants to move the box around then it should decide where the origin of the circle is. Using the position from `BoxInBox` is probably just confusing. Another reason is that these parameters are only used to initialize grid scalars, and looking at these parameters later on will give wrong information.

lucass-carneiro commented:
Thanks for the feedback! I have removed the `RESTRICTED` from `BoxInBox` and changed the toy. The boxes now move without requiring parameters from `BoxInBox`

eschnett commented:
You are currently not running the test cases in CI since you are not updating the thorn list in the `scripts` directory.

lucass-carneiro commented:
I have updated the thorn list inside `scripts`

eschnett commented:
There are no test cases...

eschnett commented on MovingBoxToy/test/circle/performance.yaml:
Don't add this file, its contents are not stable.

eschnett commented on MovingBoxToy/test/circle.par:
```diff
@@ -0,0 +1,59 @@
+CarpetX::boundary_upper_y = "dirichlet"
+CarpetX::boundary_upper_z = "dirichlet"
+
+Cactus::cctk_itlast = 20
```
20 iterations is a lot. Would 5 iterations suffice? That would make this test case 4 times smaller.

lucass-carneiro commented:
I've removed the `yml` file and reduce the size of the test
